cgsmiles/resolve.py

```diff
@@ -69,62 +68,80 @@ def match_bonding_descriptors(source, target, bond_attribute="bonding"):
 
 class MoleculeResolver:
     """
-    Resolve the molecule described by a CGBigSmile
-    string.
+    Resolve the molecule(s) described by a
+    CGSmiles string. Each execution of the
+    resolve function will return the next
+    resolved molecule and the graph of the
+    previous resolution.
+
+    Use the resolve_iter method to loop over
+    all possible molecule resolutions enconded
+    in the CGSmiles string.
     """
-
     def __init__(self,
                  pattern,
                  meta_graph=None,
                  fragment_dict={},
-                 all_atom=True):
+                 last_all_atom=True):
 
-        # let's start by setting some attributes
-        # this is the fragment string
-        self.fragment_str = None
-        # this is the cgsmiles string
-        self.cgsmiles_str = None
+        """
+        Parameters
+        ----------
+        pattern: str
+            the cgsmiles string to resolve
+        meta_graph: `:class:nx.Graph`
+            a potential higher resolution graph
+        fragment_dict: dict[`:class:nx.Garph`]
+            a dict of fragment graphs
```

Comment:
```suggestion
        fragment_dict: dict[str, nx.Graph]
            a dict of fragment graphs
```
Assuming it maps name to graph